polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -2332,13 +2141,6 @@ ValueCategory MLIRScanner::EmitScalarCast(mlir::Location Loc, ValueCategory Src,
 ValueCategory MLIRScanner::EmitScalarConversion(ValueCategory Src,
                                                 QualType SrcQT, QualType DstQT,
                                                 SourceLocation Loc) {
-  // TODO: By a flaw in how operators are handled, SrcQT or DstQT might be
-  // non-scalar types. Remove when we actually call operatorOP member functions.
-  if (!(SrcQT.isNull() || SrcQT.isCanonical()) ||
-      !(DstQT.isNull() || DstQT.isCanonical())) {
-    return Src;
-  }
```

Comment:
Not needed anymore.

Reply:
I have a draft PR that essentially removed this to fix an assert in one of the Rodinia benchmarks. Is this fix unrelated to this changes ?  It would seems so, and if that is the case should we continue with PR https://github.com/intel/llvm/pull/7752

Reply:
This fixes several cast problems, as that one. I can rebase on your work when it's merged if you want.